File: open-sdk-java/src/main/java/com/youzan/open/sdk/api/AbstractAPI.java
package com.youzan.open.sdk.api;

import com.youzan.open.sdk.model.APIParams;

/**
 * @author ph0ly
 * @time 2016-11-28
 */
public abstract class AbstractAPI implements API {

    protected APIParams params;

    public APIParams getAPIParams() {
        return this.params;
    }

    public void setAPIParams(APIParams apiParams) {
        this.params = apiParams;
    }

    public String getHttpUrl() {
        return "http://open.koudaitong.com/api";
    }

    public String getVersion() {
        return "1.0.0";
    }
}

File: api/AbstractAPI.cs
using System;
using System.Collections.Generic;
using open_sdk.common.constant;

namespace open_sdk.api
{
    public abstract class AbstractAPI : IAPI
    {
        protected IApiParams apiParams;

        protected string gateway = "https://open.youzanyun.com";

        protected string version;

        protected Dictionary<string, string> headers = new Dictionary<string, string>();

        public void AddHeader(string headerName, string headerValue)
        {
            headers.Add(headerName, headerValue);
        }

        public IApiParams GetAPIParams()
        {
            return apiParams;
        }

        public void SetAPIParams(IApiParams apiParams)
        {
            this.apiParams = apiParams;
        }

        public void SetGateway(string gateway)
        {
            this.gateway = gateway;
        }

        public IDictionary<string, string> GetHeaders()
        {
            return headers;
        }

        public string GetVersion()
        {
            return version;
        }

       
        public abstract string GetGateway();
        public abstract OAuthEnum GetOAuthType();
        public abstract string GetHttpMethod();
        public abstract string GetName();

       
    }
}
